sdk/testutil/testlog.go

```diff
@@ -15,6 +17,16 @@ func init() {
 	sendTestLogsToStdout = os.Getenv("NOLOGBUFFER") == "1"
 }
 
+func TestHcLog(name string) *log.Logger {
+	consulLogger := hclog.New(&hclog.LoggerOptions{
+		Name:  name,
+		Level: log.LstdFlags,
+	})
+	return consulLogger.StandardLogger(&hclog.StandardLoggerOptions{
+		InferLevels: true,
+	})
+}
+
```

Comment:
I think we may be better served if the old methods return `*log.Logger` and the new methods return `hclog.Logger` and if you need both we can add a helpful shim like:
```
// in practice the symbol would still retain "test" in the name as "testutil.Logger" so we
// avoid the weird mixed-casing of "hclog" and keep the "nice" naming of the function.
func Logger(t testing.TB) hclog.Logger {
  return hclog.New(&hclog.LoggerOptions{
    Name: t.Name(),
    Level: log.LstdFlags,
    Output: &testWriter{t},
  }
}

// in practice the symbol would still retain "test" in the name as "testutil.LoggerWithName" so we
// avoid the weird mixed-casing of "hclog" and keep the "nice" naming of the function.
func LoggerWithName(t testing.TB, name string) hclog.Logger {
  return hclog.New(&hclog.LoggerOptions{
    Name: "test["+name+"]",
    Level: log.LstdFlags,
    Output: &testWriter{t},
  }
}

func LogShim(logger hclog.Logger) *log.Logger { ... just the StandardLogger thing ... }

// Deprecated: use Logger
func TestLogger(t testing.TB) *log.Logger { ... }

// Deprecated: use LoggerWithName
func TestLoggerWithName(t testing.TB, name string) *log.Logger { ... }
```